components/Controllers/OpenVault/VaultWriteButton.tsx

```diff
@@ -143,8 +145,27 @@ export function VaultWriteButton({
 
   const buttonText = useMemo(() => {
     if (!oracleInfoSynced) return 'Waiting for oracle...';
-    return vaultHasDebt ? 'Update Loan' : 'Borrow';
-  }, [oracleInfoSynced, vaultHasDebt]);
+    const updateOrBorrowText =
```

Comment:
think we can maybe cleanup
```suggestion
    if (depositNFT.length == 0 && withdrawNFTs.length> 0) {
     "withdraw"
     } else if (depositNFT.length > 0 && withdrawNFTs.length == 0 && amount.isZero()) {
     "deposit"
     } else if (depositNFT.length > 0 && withdrawNFTs.length == 0 && !amount.isZero()) {
     "Borrow"
     } else {
     "Update Loan"
     }
```

Reply:
need amount.isZero() on the first if statement though? lots of redundant amount checks imo

Reply:
think its better to separate by whether amount is zero or not